meta-balena-common/recipes-support/balena-config-vars/balena-config-vars/os-config-json

```diff
@@ -0,0 +1,98 @@
+#!/bin/sh
+#
+# Copyright 2022 Balena Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+if [ -f "/usr/libexec/os-helpers-logging" ]; then
+	. /usr/libexec/os-helpers-logging
+else
+	error() {
+		echo "[ERROR] $*"
+	}
+	info() {
+		echo "[INFO] $*"
+	}
+fi
+
+CONF_TMPDIR=${CONF_TMPDIR:-"/var/volatile"}
+CONFIG_PATH=${CONFIG_PATH:-"/mnt/boot/config.json"}
+UNIT_CONF_PATH=${UNIT_CONF_PATH:-"/etc/systemd/unit-conf.json"}
+
+if [ ! -f "${UNIT_CONF_PATH}" ]; then
+	error "${UNIT_CONF_PATH} not found"
+	exit 1
+fi
+if [ ! -f "${CONFIG_PATH}" ]; then
+	error "${CONFIG_PATH} not found"
+	exit 1
+fi
+
+# Exit if syntactically incorrect
+jq -e '.' "${CONFIG_PATH}" > /dev/null
+jq -e '.' "${UNIT_CONF_PATH}" > /dev/null
+
+mkdir -p "${CONF_TMPDIR}"
+
+_tmpdir=$(mktemp -d)
+units=$(jq -r '.properties.units.properties | keys[]' "${UNIT_CONF_PATH}")
+for unit in ${units}; do
+	confs=$(jq -r ".properties.units.properties | .[\"${unit}\"].configuration" "${UNIT_CONF_PATH}")
+	for conf in ${confs}; do
+		_path=${conf%.*}
+		_value=${conf##*.}
+		if [ "${_path}" = "${_value}" ]; then
+			_path=""
+		fi
+		_entry=$(jq ".${_path} | to_entries | [.[] | select(.key == \"${_value}\")] | from_entries" "${CONFIG_PATH}" 2> /dev/null || true)
+		if [ -n "${_entry}" ] && [ "${_entry}" != "{}" ]; then
+			if [ -f "${_tmpdir}/${unit}.json" ]; then
+				_json=$(jq --argjson entry "${_entry}" '. += $entry' "${_tmpdir}/${unit}.json")
+				echo "${_json}" > "${_tmpdir}/${unit}.json"
+			else
+				echo "${_entry}" > "${_tmpdir}/${unit}.json"
+			fi
+		elif [ "${_entry}" == "{}" ]; then
+			rm -f "${_tmpdir}/${unit}.json"
+
+		fi
+	done
+	if [ -f "${_tmpdir}/${unit}.json" ]; then
+		if [ -f "${CONF_TMPDIR}/${unit}.json" ]; then
+			_newcksum=$(md5sum "${_tmpdir}/${unit}.json" | cut -d " " -f1)
+			_oldcksum=$(md5sum "${CONF_TMPDIR}/${unit}.json" | cut -d " " -f1)
+			if [ "${_newcksum}" != "${_oldcksum}" ]; then
+				info "${unit} configuration changed"
+				mv "${_tmpdir}/${unit}.json" "${CONF_TMPDIR}/${unit}.json"
+			fi
+		else
+			info "New ${unit} configuration"
+			touch "${CONF_TMPDIR}/${unit}.json"
+			if find "/lib/systemd/system" -name "${unit}-conf.path" | grep -q "${unit}-conf.path"; then
```

Comment:
```suggestion
			if systemctl list-units --full -all | grep -Fq "${unit}-conf.path"; then
```
This find seems clunky to me, there are a number of systemctl commands that could do the job better.

This one seems to be reliable and is used for bash-completion: https://stackoverflow.com/a/53640320

Reply:
thanks @klutchell  I'll review it. I was using `systemctl list-units --all` initially, but I think it did not list path units if the file they were watching does not exist. So I used the find.
Now I realize that later I resorted to touching the file to watch before so it may work now.

Reply:
Updated.